src/BinSkim.Rules/PERules/BA2025.EnableShadowStack.cs

```diff
@@ -0,0 +1,96 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.Composition;
+using System.Reflection.PortableExecutable;
+
+using Microsoft.CodeAnalysis.BinaryParsers;
+using Microsoft.CodeAnalysis.BinaryParsers.PortableExecutable;
+using Microsoft.CodeAnalysis.IL.Sdk;
+using Microsoft.CodeAnalysis.Sarif;
+using Microsoft.CodeAnalysis.Sarif.Driver;
+
+namespace Microsoft.CodeAnalysis.IL.Rules
+{
+    [Export(typeof(Skimmer<BinaryAnalyzerContext>)), Export(typeof(ReportingDescriptor)), Export(typeof(IOptionsProvider))]
+    public class EnableShadowStack : WindowsBinaryAndPdbSkimmerBase
+    {
+        private const int IMAGE_DEBUG_TYPE_EX_DLLCHARACTERISTICS = 20;
+        private const ushort IMAGE_DLLCHARACTERISTICS_EX_CET_COMPAT = 0x001;
+
+        /// <summary>
+        /// BA2025
+        /// </summary>
+        public override string Id => RuleIds.EnableShadowStack;
+
+        /// <summary>
+        /// Control-flow Enforcement Technology (CET) Shadow Stack is a computer processor feature
+        /// that provides capabilities to defend against return-oriented programming (ROP) based
+        /// malware attacks.
+        /// </summary>
+        public override MultiformatMessageString FullDescription => new MultiformatMessageString
+        {
+            Text = RuleResources.BA2025_EnableShadowStack_Description
+        };
+
+        protected override IEnumerable<string> MessageResourceNames => new string[] {
+            nameof(RuleResources.BA2025_Pass),
+            nameof(RuleResources.BA2025_Warning),
+            nameof(RuleResources.NotApplicable_InvalidMetadata)
+        };
+
+        public override AnalysisApplicability CanAnalyzePE(PEBinary target, Sarif.PropertiesDictionary policy, out string reasonForNotAnalyzing)
+        {
+            PE portableExecutable = target.PE;
+            AnalysisApplicability result = AnalysisApplicability.NotApplicableToSpecifiedTarget;
+
+            reasonForNotAnalyzing = MetadataConditions.ImageIsILOnlyAssembly;
+            if (portableExecutable.IsILOnly) { return result; }

```

Comment:
other way is to move the assign statement inside the 
if() {assign statement;return result; }
since it is only needed when if is true

Reply:
pls, do the proper changes :)

Reply:
got it, updated.